src/hooks/http.test.js

```diff
@@ -493,6 +502,157 @@ describe('http hook', () => {
+    };
+
+    HttpsScenarioBuilder.appendNextResponse(responseData.body);
+    const wrappedRequest = httpHook.httpRequestWrapper(HttpsMocker.request);
+
+    wrappedRequest(requestData, () => {});
+
+    const spans = SpansContainer.getSpans();
+
+    const expectedSpan = new HttpSpanBuilder()
+      .withStarted(spans[0].started)
+      .withEnded(spans[0].ended)
+      .withSpanId(spans[0].id)
+      .withHttpInfo({
+        host: HttpSpanBuilder.DEFAULT_HOST,
+        request: requestData,
+        response: responseData,
+      })
+      .withRequestTimesFromSpan(spans[0])
+      .build();
+
+    expect(spans).toEqual([expectedSpan]);
+  });
+
+  test('httpRequestWrapper - added span before request finish', () => {
+    utils.setTimeoutTimerEnabled();
+    TracerGlobals.setHandlerInputs({
+      event: {},
+      context: {
+        awsRequestId: HttpSpanBuilder.DEFAULT_PARENT_ID,
+        invokedFunctionArn: HttpSpanBuilder.DEFAULT_ARN,
+        functionVersion: HttpSpanBuilder.DEFAULT_VERSION,
+      },
+    });
+    const requestData = HttpSpanBuilder.DEFAULT_REQUEST_DATA;
+
+    HttpsScenarioBuilder.dontFinishNextRequest();
+    const wrappedRequest = httpHook.httpRequestWrapper(HttpsMocker.request);
+
+    wrappedRequest(requestData, () => {});
+
+    const spans = SpansContainer.getSpans();
+
+    const expectedSpan = new HttpSpanBuilder()
+      .withStarted(spans[0].started)
+      .withEnded(spans[0].ended)
+      .withSpanId(spans[0].id)
+      .withHttpInfo({
+        host: HttpSpanBuilder.DEFAULT_HOST,
+        request: requestData,
+      })
+      .withNoResponse()
+      .withRequestTimesFromSpan(spans[0])
+      .build();
+
+    expect(spans).toEqual([expectedSpan]);
+  });
+
+  test('httpRequestWrapper - wrapping twice not effecting', () => {
+    TracerGlobals.setHandlerInputs({
+      event: {},
+      context: {
+        awsRequestId: HttpSpanBuilder.DEFAULT_PARENT_ID,
+        invokedFunctionArn: HttpSpanBuilder.DEFAULT_ARN,
+        functionVersion: HttpSpanBuilder.DEFAULT_VERSION,
+      },
+    });
+    const requestData = HttpSpanBuilder.DEFAULT_REQUEST_DATA;
+    const responseData = {
+      statusCode: 200,
+      body: 'OK',
+    };
+
+    HttpsScenarioBuilder.appendNextResponse(responseData.body);
+    const wrappedRequest = httpHook.httpRequestWrapper(HttpsMocker.request);
+    const wrappedRequestTwice = httpHook.httpRequestWrapper(wrappedRequest);
+
+    wrappedRequestTwice(requestData, () => {});
+
+    const spans = SpansContainer.getSpans();
+
+    const expectedSpan = new HttpSpanBuilder()
+      .withStarted(spans[0].started)
+      .withEnded(spans[0].ended)
+      .withSpanId(spans[0].id)
+      .withHttpInfo({
+        host: HttpSpanBuilder.DEFAULT_HOST,
+        request: requestData,
+        response: responseData,
+      })
+      .withRequestTimesFromSpan(spans[0])
+      .build();
+
+    expect(spans).toEqual([expectedSpan]);
+  });
+
+  test('httpRequestWrapper - circular object', () => {
```

Comment:
Add 2 words explaining what should happen with a circular object.

Reply:
We are getting the headers as JSON from our customers, they can do whatever they want and all standards HTTP client support this shit, so we need also to not cursing on stack overflow (This test added with blood ;) )

Reply:
Added explain